Application verification fails on ios 15 with an older version of Xcode

#### Issue details
Starting with iOS 15, the system checks a new signature [format](https://developer.apple.com/documentation/xcode/using-the-latest-code-signature-format) (DER) so that applications can run on current OS.
For lower versions of XCode 12.5 it is necessary to include the generate-entitlement-der flag.

Robovm must sign with `generate-entitlement-der` flag for older XCode versions.

#### Reproduction steps/code

Build and run a app in real device with iOS15 in Xcode 12.4

##### Robovm output
```shell
Caused by: org.robovm.libimobiledevice.LibIMobileDeviceException: ApplicationVerificationFailed
        at org.robovm.libimobiledevice.util.AppLauncher$1.error(AppLauncher.java:1006)
        at org.robovm.libimobiledevice.Callbacks.callInstproxyCallback(Callbacks.java:64)
```
##### ideviceinstaller output
```shell
Install: CreatingStagingDirectory (5%)
Install: ExtractingPackage (15%)
Install: InspectingPackage (20%)
Install: TakingInstallLock (20%)
Install: PreflightingApplication (30%)
Install: InstallingEmbeddedProfile (30%)
Install: VerifyingApplication (40%)ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008029: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.19Noon/extracted/myapp.app : 0xe8008029 (The code signature version is no longer supported.)
```

check signature:
```shell
codesign -dvvvvv myapp.app
```
I check under Page size in the signature and -5 and -7 values is not present.

#### Configuration

**Build Tools:**
- Gradle plugin
 
**Versions:**
- Robovm: 2.3.14
- XCode: 12.4
- JDK: 11
- Mac mini 2012
- XCode 12.4
- macOS Catalina

**Build Targets:**
- iPhone XR device (iOS 15)

---
### References
https://www.browserstack.com/question/917
https://developer.apple.com/forums/thread/682775?page=2
https://stackoverflow.com/questions/68467306/the-code-signature-version-is-no-longer-supported

--- a/compiler/compiler/src/main/java/org/robovm/compiler/target/ios/IOSTarget.java
+++ b/compiler/compiler/src/main/java/org/robovm/compiler/target/ios/IOSTarget.java
@@ -639,7 +639,15 @@ public class IOSTarget extends AbstractTarget {
         codesign(identity, entitlementsPList, false, false, true, extensionDir);
     }
 
-    private void codesign(SigningIdentity identity, File entitlementsPList, boolean preserveMetadata, boolean verbose, boolean allocate, File target) throws IOException {
+    private void codesign(SigningIdentity identity, File entitlementsPList, boolean preserveMetadata,
+                          boolean verbose, boolean allocate, File target) throws IOException {
+        // just a wrapper that forces "--generate-entitlement-der" for all kind of signing
+        boolean generateDerEntitlement = true;
+        codesign(identity, entitlementsPList, preserveMetadata, generateDerEntitlement, verbose, allocate, target);
+    }
+
+    private void codesign(SigningIdentity identity, File entitlementsPList, boolean preserveMetadata,
+                          boolean generateDerEntitlement, boolean verbose, boolean allocate, File target) throws IOException {
         List<Object> args = new ArrayList<>();
         args.add("-f");
         args.add("-s");
@@ -650,6 +658,9 @@ public class IOSTarget extends AbstractTarget {
         }
         if (preserveMetadata) {
             args.add("--preserve-metadata=identifier,entitlements");
+        }
+        if (generateDerEntitlement) {
+            args.add("--generate-entitlement-der");
         }
         if (verbose) {
             args.add("--verbose");

